tiatoolbox/models/dataset/dataset_abc.py

```diff
@@ -143,11 +145,14 @@ def preproc_func(self, func):
         elif callable(func):
             self._preproc = func
         else:
-            raise ValueError(f"{func} is not callable!")
+            msg = f"{func} is not callable!"
+            raise ValueError(msg)
 
-    def __len__(self):
+    def __len__(self) -> int:
+        """Returns the length of the instance attributes."""
         return len(self.inputs)
 
     @abstractmethod
     def __getitem__(self, idx):
+        """Defines the behaviour when an item is accessed."""
```

Comment:
```suggestion
        """Get an item from the dataset."""
```

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/625/commits/a24a4918731addef5dc1da96c9fe6e396b122a37